charts/kyuubi/values.yaml

```diff
@@ -57,12 +57,60 @@ probe:
     failureThreshold: 10
     successThreshold: 1
 
-server:
-  bind:
-    host: 0.0.0.0
+# Kyuubi frontends
+frontend:
```

Comment:
@pan3793 Happy holidays!

Thank you for the explanation above. Let's discuss the schema for `values.yaml` file:

1. The reason why I moved `frontend` configs (`thriftBinary`, `thriftHttp`, `rest` etc) outside of `kyuubi` conf is because they configure not only Kyuubi, but also Kubernetes objects on top of it. I.e. they configure how users interact with Kyuubi application in Kubernetes. In my opinion, from helm chart user point of view, frontends are dedicated integration points for Kyuubi and should be placed separately.
I can rename `frontend` to `service`. I just used `frontend` word, because it is used in the [doc](https://kyuubi.readthedocs.io/en/master/deployment/settings.html#frontend), and to explicitly say what kind of services these values are for.

2. Also I made `kyuubi` configs flat (`kyuubiConfDir` and `kyuubiConf`) following this recommendation ["flat-or-nested-values"](https://helm.sh/docs/chart_best_practices/values/#flat-or-nested-values), and want to do similar for `spark` (`sparkConfDir` and `sparkConf`). I'm not sure whether it is good for the future, but it seems to be a relevant choice for now. To be clear, I think it's also good to keep it like
    ```
    kyuubi
      confDir: /opt/kyuubi/conf
      conf:
        kyuubiDefaults: ~
        kyuubiEnv: ~
    
    spark:
      confDir: /opt/spark/conf
      conf:
        sparkDefaults: ~
        sparkEnv: ~
    ```

My personal preference for now is the following schema:
```
frontend:
  thriftBinary:
      xxxxx
  thriftHttp:
    xxxxx
  rest:
    xxxxx
  mysql:
    xxxxx
  trino:
    xxxxx

kyuubiConfDir: /opt/kyuubi/conf
kyuubiConf:
  kyuubiEnv: ~
  kyuubiDefaults: ~
  log4j2: ~

sparkConfDir: /opt/spark/conf
sparkConf:
  kyuubiEnv: ~
  sparkDefaults: ~
```

Reply:
@dnskr I prefer the nested style here since `kyuubi`/`spark` is a kind of "namespace"/"component", not the leaf nodes as the example.

BTW, I found the similar usage in superset
https://github.com/apache/superset/blob/master/helm/superset/values.yaml#L732

Reply:
The example you refer relates to `redis` subchart config, which is additional component which can or cannot be deployed by superset chart.

Reply:
spark, flink, hive are optional too. suppose the user only wants to use flink engine, then he/she don't need any spark conf

Reply:
I get your point, they are subchart

Reply:
OK, flat style is fine to me, how about changing "frontend" to "listen"? do you think it's easy to understand for users?

Reply:
> May be `service`? I didn’t see `listen` is used in other charts.

Bad idea, because there is nested `service` tree

Reply:
cc @yaooqinn, do you have suggestions here?

Reply:
@pan3793 `api` or `interface` maybe?

Reply:
If no good choice, let's keep "server" instead

Reply:
done